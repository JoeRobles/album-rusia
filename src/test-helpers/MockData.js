import moment from 'moment';

export const expensesReducerDefaultState = [
  {
    id: 'poijasdfhwer',
    description: 'January Rent',
    note: 'This was the final payment for that address',
    amount: 54500,
    createdAt: 1514954400000,
  },
  {
    id: 'lmdfmlgkdnfg',
    description: 'water bill',
    note: '',
    amount: 100,
    createdAt: 1515085200000,
  },
  {
    id: 'fhgerthhrdhd',
    description: 'gas bill',
    note: '',
    amount: 300,
    createdAt: 1515140400000,
  },
  {
    id: 's√±ojglsjslkg',
    description: 'internet bill',
    note: '',
    amount: 200,
    createdAt: 1515267500000,
  },
];

export const filtersReducerDefaultState = {
  text: '',
  sortBy: 'date',
  asc: true,
  startDate: moment().startOf('month'),
  endDate: moment().endOf('month'),
};

export const filtersReducerMockState = {
  text: 'bills',
  sortBy: 'amount',
  asc: true,
  startDate: moment(0),
  endDate: moment(0).add('3', 'days'),
};
