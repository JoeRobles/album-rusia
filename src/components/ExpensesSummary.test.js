import React from 'react';
import { shallow } from 'enzyme';
import { expensesReducerDefaultState } from '../test-helpers/MockData';
import {ExpensesSummary} from './ExpensesSummary';

it('should correctly render ExpensesSummary with 1 expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235}/>);
  expect(wrapper).toMatchSnapshot();
});

it('should correctly render ExpensesSummary with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={23512340987}/>);
  expect(wrapper).toMatchSnapshot();
});