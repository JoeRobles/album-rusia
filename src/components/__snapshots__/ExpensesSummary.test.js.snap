// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `
<tfoot>
  <tr>
    <th>
      Viewing
    </th>
    <th>
       
      1
       
      expense
    </th>
    <th>
      totalling 
      $2.35
    </th>
  </tr>
</tfoot>
`;

exports[`should correctly render ExpensesSummary with multiple expenses 1`] = `
<tfoot>
  <tr>
    <th>
      Viewing
    </th>
    <th>
       
      23
       
      expenses
    </th>
    <th>
      totalling 
      $235,123,409.87
    </th>
  </tr>
</tfoot>
`;
